cairo/ethereum/cancun/state.cairo

```diff
@@ -303,6 +310,52 @@ func set_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
     return ();
 }
 
+func destroy_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(address: Address) {
+    alloc_locals;
+
+    let storage_tries = state.value._storage_tries;
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    let storage_tries_dict_ptr = cast(storage_tries.value.dict_ptr, DictAccess*);
+
+    // Use `hashdict_get` instead of `hashdict_read` because `MappingAddressTrieBytes32U256` is not a
+    // `default_dict`. Accessing a key that does not exist in the dict would have panicked for `hashdict_read`.
+    let (pointer) = hashdict_get{poseidon_ptr=poseidon_ptr, dict_ptr=storage_tries_dict_ptr}(
+        1, &address.value
+    );
+
+    if (pointer == 0) {
+        return ();
+    }
```

Comment:
a dict_get writes in the dict's memory segment. you need to rebind the storage_tries_dict_ptr before returning